Table Dependency: Not Found with deriveSchemaFromMigrations

### SQLDelight Version

2.0.0-SNAPSHOT

### Operating System

macOS 13

### Gradle Version

7.4.2

### Kotlin Version

1.7.0

### Dialect

PostgreSQL

### AGP Version

_No response_

### Describe the Bug

Module `consumer` depends on the database declared in module `db-schema`. This db uses `deriveSchemaFromMigrations=true`. With this option sqldelight is unable to find the tables from `db-schema` in the query files defined in `consumer`. 

Reproducer: [untitled2.zip](https://github.com/cashapp/sqldelight/files/8892677/untitled2.zip)
 

### Stacktrace

```shell
Compiling with dialect app.cash.sqldelight.dialects.postgresql.PostgreSqlDialect

/Users/philipwedemann/Downloads/untitled2/consumer/src/main/sqldelight/app/consumer/Foo.sq: (2, 14): No table found with name foo
2    SELECT * FROM foo
                   ^^^


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':consumer:generateMainRawDBInterface'.
> A failure occurred while executing app.cash.sqldelight.gradle.SqlDelightTask$GenerateInterfaces
   > Generation failed; see the generator error output for details.
```


### Gradle Build Script

```gradle
// db-schema
plugins {
    kotlin("jvm")
    id("app.cash.sqldelight")
}

sqldelight {
    database("RawDB") {
        dialect("app.cash.sqldelight:postgresql-dialect:2.0.0-SNAPSHOT")
        packageName = "app.schema"
        deriveSchemaFromMigrations = true
    }
}

// consumer
plugins {
    kotlin("jvm")
    id("app.cash.sqldelight")
}

sqldelight {
    database("RawDB") {
        dialect("app.cash.sqldelight:postgresql-dialect:2.0.0-SNAPSHOT")
        dependency(projects.dbSchema)
        packageName = "app.consumer"
    }
}
```

--- a/docs/common/gradle-dependencies.md
+++ b/docs/common/gradle-dependencies.md
@@ -22,3 +22,4 @@ sqldelight {
   }
 }
 ```
+If you use `deriveSchemaFromMigrations = true`, every module must also enable this feature.

--- a/docs/common/gradle-dependencies.md
+++ b/docs/common/gradle-dependencies.md
@@ -22,4 +22,4 @@ sqldelight {
   }
 }
 ```
-If you use `deriveSchemaFromMigrations = true`, every module must also enable this feature.
+If you use `deriveSchemaFromMigrations = true`, every module depending on this module must also enable this feature.

